src/java/org/apache/cassandra/index/sai/plan/StorageAttachedIndexSearcher.java

```diff
@@ -383,7 +404,7 @@ private UnfilteredRowIterator queryStorageAndFilter(PrimaryKey key)
             }
         }
 
-        private UnfilteredRowIterator applyIndexFilter(PrimaryKey key, UnfilteredRowIterator partition, FilterTree tree)
```

Comment:
nit: how about `keysInTheSamePartition`

Reply:
Possibly. Let's see what happens in `applyIndexFilter()` below. There might be a version of that that doesn't require the list of keys as an argument...